daphne_worker/src/config.rs

```diff
@@ -838,8 +856,23 @@ impl<'srv> DaphneWorker<'srv> {
     pub(crate) async fn worker_request_to_dap(
         &self,
         mut req: Request,
-    ) -> Result<DapRequest<BearerToken>> {
-        let sender_auth = req.headers().get("DAP-Auth-Token")?.map(BearerToken::from);
+    ) -> Result<DapRequest<DaphneWorkerAuth>> {
+        // Determine the authorization method used by the sender.
+        let bearer_token = req.headers().get("DAP-Auth-Token")?.map(BearerToken::from);
+        let tls_client_auth = req.cf().tls_client_auth();
+        let sender_auth = match (bearer_token, tls_client_auth) {
+            (Some(bearer_token), None) => Some(DaphneWorkerAuth::BearerToken(bearer_token)),
+            (None, Some(tls_client_auth)) => Some(DaphneWorkerAuth::CfTlsClientAuth {
```

Comment:
We pull out the issuer and subject so that we can check they have the expected values. Arguably it's sufficient to just check that a certificate is present (i.e., `tls_client_auth.is_some()`) and let the zone configuration bear the burden of checking these names. My goal here is to add a bit of resilience in case the zone config gets clobbered somehow.

Note that the edge metal is expected to handle the rest of the details of certificate validation, including checking the chain and signature, enforcing lifetime, etc.

Reply:
Thinking about this more: Perhaps it's best to just check that a certificate is present. One problem that might come up is changing, say, the subject name when rotating the certificate: We will have to synchronize that rotation with deploying the new change in the environment variable to the Worker. Would it be better to just let the zone config handle everything?

Reply:
If we keep this feature, perhaps it would be best to make this a list of issuer/subject pairs that we trust.

Reply:
As discussed offline: We have kept the expected issuer, but there is now a list of valid "subjects".